manifests/fragment.pp

```diff
@@ -15,27 +15,19 @@
 #   anything else using this to influence the order of the content in the file
 #
 define concat::fragment(
-    $target,
-    $ensure  = undef,
-    $content = undef,
-    $source  = undef,
-    $order   = '10',
+    Variant[String, Stdlib::Compat::String]                                   $target,
+                                                                              $ensure  = undef,
+    Optional[Variant[String, Stdlib::Compat::String]]                         $content = undef,
+    Optional[Variant[String, Stdlib::Compat::String, Array]]                  $source  = undef,
+    Variant[String, Stdlib::Compat::String, Integer, Stdlib::Compat::Integer] $order   = '10',
 ) {
-  validate_string($target)
```

Comment:
Instead of straight up removing these, use the validate_legacy function here: https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/functions/validate_legacy.rb
A description of why/usage: https://github.com/puppetlabs/puppetlabs-stdlib#validate_legacy